graylog2-server/src/main/java/org/graylog/plugins/sidecar/rest/resources/ConfigurationResource.java

```diff
@@ -124,6 +130,29 @@ public ConfigurationListResponse listConfigurations(@ApiParam(name = "page") @Qu
         return ConfigurationListResponse.create(query, configurations.pagination(), total, sort, order, result);
     }
 
+    @GET
+    @Path("/uploads")
+    @RequiresPermissions(SidecarRestPermissions.CONFIGURATIONS_READ)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "List all uploaded configurations")
+    public CollectorUploadListResponse listImports(@ApiParam(name = "page") @QueryParam("page") @DefaultValue("1") int page,
+                                                   @ApiParam(name = "per_page") @QueryParam("per_page") @DefaultValue("10") int perPage,
+                                                   @ApiParam(name = "query") @QueryParam("query") @DefaultValue("") String query,
+                                                   @ApiParam(name = "sort",
+                                                           value = "The field to sort the result on",
+                                                           required = true,
+                                                           allowableValues = "id,collector_id, created")
+                                                   @DefaultValue(Configuration.FIELD_NAME) @QueryParam("sort") String sort,
+                                                   @ApiParam(name = "order", value = "The sort direction", allowableValues = "asc, desc")
+                                                   @DefaultValue("asc") @QueryParam("order") String order) {
+        final SearchQuery searchQuery = searchQueryParser.parse(query);
```

Comment:
We don't use any queries and this parser is not made for that collection. We also don't use sorting or ordering. I think we should remove all things we will never use from here, even if this method will most likely go away in some months.